components/datepicker/DatePicker.js

```diff
@@ -9,24 +9,15 @@ import theme from './theme.css';
 
 class DatePicker extends PureComponent {
   state = {
-    selectedDate: null,
+    selectedDate: this.props.selectedDate,
   };
 
-  componentWillMount() {
-    const { selectedDate } = this.props;
-
-    if (selectedDate) {
-      this.setState({ selectedDate });
-    }
+  componentDidUpdate() {
+    this.props.onChange({ selectedDate: this.state.selectedDate });
```

Comment:
When is the `componentDidUpdate` callback triggered? Say a parent component passes a new `selectedDate` to the datePicker component. Once the datePicker has re-rendered based on that `prop` change won't it trigger `componentDidUpdate` and thus also trigger the changeHandler passed from the parent even though the value didn't change?

Reply:
it's triggered on render

Reply:
I `eliminated` componentDidUpdate because we only want `this.props.onChange` to be called when a user actually clicks on a day.

 :ok: by @driesd on 30/05/2018 16:39:04